src/pages/index.js

```diff
@@ -91,53 +75,106 @@ function renderCard(data) {
 const imagePopup = new PopupWithImage("#preview-image-modal");
 imagePopup.setEventListeners();
 
-//FORM POPUP
+//DELETE CARD POPUP
+const deleteCardPopup = new PopupWithConfirmation("#card__delete-modal");
+deleteCardPopup.setEventListeners();
+
+const profileUpdateSubmit = new PopupWithConfirmation("#profile-edit-modal");
+const newCardSubmit = new PopupWithConfirmation("#add-new-card");
+
+function handleDeleteClick(card) {
+  deleteCardPopup.open();
+  deleteCardPopup.setSubmitAction(() => {
+    deleteCardPopup.setSubmitText(true, "Deleting...");
+    api
+      .deleteCard(card.id)
+      .then(() => {
+        deleteCardPopup.close();
+        card.removeCard();
+      })
+      .catch((err) => console.log(err))
+      .finally(() => deleteCardPopup.setSubmitText(false));
+  });
+}
 
+//FORM POPUP
 const openImagePopup = new PopupWithForm(
   "#add-new-card",
   handleImageFormSubmit
 );
-const editProfile = new PopupWithForm(
-  "#profile-edit-modal",
-  handleFormSubmit
+const editProfile = new PopupWithForm("#profile-edit-modal", handleFormSubmit);
+const profilePicEdit = new PopupWithForm(
+  "#profile__image-edit",
+  profilePicEditSubmit
 );
+
 editProfile.setEventListeners();
+profilePicEdit.setEventListeners();
 openImagePopup.setEventListeners();
 
 const editFormValidator = new FormValidator(config, profileModalForm);
+const profilePicValidator = new FormValidator(config, profilePicModal);
 const cardFormValidator = new FormValidator(config, addNewCardModal);
 
 editFormValidator.enableValidation();
+profilePicValidator.enableValidation();
 cardFormValidator.enableValidation();
 
 //FORM SUBMIT
 function handleFormSubmit(data) {
-  console.log(data)
-  userInfo.setUserInfo(data)
-  editProfile.close();
+  api.profileUpdate(data).then(() => {
```

Comment:
**Could be improved**

If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **closing** and **catching errors**


```js

// You can make a universal function that accepts a request, popup instance and optional loading text  
function handleSubmit(request, popupInstance, loadingText = "Saving...") {

  // here we change the button text
  popupInstance.renderLoading(true, loadingText);
  request()
    .then(() => {
      // We need to close only in `then`
      popupInstance.close()
    })
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
     .catch(console.error)
    // in `finally` we need to return the initial button text back in any case
    .finally(() => {
      popupInstance.renderLoading(false);
    });
}



```


You need to pass the **request** into the call of `handleSubmit` like that:



```js

// here is an example of the profile form handling
function handleProfileFormSubmit(inputValues) {
  // we create a function that returns a promise
  function makeRequest() {
    // `return` lets us use a promise chain `then, catch, finally` inside `handleSubmit`
    return api.editProfile(inputValues).then((userData) => {
      userInfo.setUserInfo(userData)
    });
  }
  // Here we call the function passing the request, popup instance and if we need some other loading text we can pass it as the 3rd argument
  handleSubmit(makeRequest, profilePopup);
}



```